bigchaindb/pipelines/block.py

```diff
@@ -111,28 +112,27 @@ def write(self, block):
         """Write the block to the Database.
 
         Args:
-            block (dict): the block of transactions to write to the database.
+            block (Block): the block of transactions to write to the database.
 
         Returns:
             The block.
         """
-        logger.info('Write new block %s with %s transactions',
-                    block['id'],
-                    len(block['block']['transactions']))
+        logger.info('Write new block {} with {} transactions'.format(block.id,
+                    len(block.transactions)))
         self.bigchain.write_block(block)
         return block
 
     def delete_tx(self, block):
         """Delete transactions.
 
         Args:
-            block (dict): the block containg the transactions to delete.
+            block (Block): the block containg the transactions to delete.
 
         Returns:
             The block.
         """
         r.table('backlog')\
-         .get_all(*[tx['id'] for tx in block['block']['transactions']])\
+         .get_all(*[tx['id'] for tx in block.to_dict()['block']['transactions']])\
```

Comment:
I didn't notice that this function was part of `pipelines/block.py`.
We'll have to return the block. I fixed it by not deserializing the block.